client/my-sites/patterns/components/pattern-gallery/client.tsx

```diff
@@ -14,12 +15,68 @@ import './style.scss';
 
 const LOGGED_OUT_USERS_CAN_COPY_COUNT = 3;
 
+type MasonryGalleryProps = PropsWithChildren< {
+	className?: string;
+	enableMasonry: boolean;
+} >;
+
+// Simulates a Masonry layout by applying negative `margin-top` on every item that doesn't sit in
+// the first row
+function MasonryGallery( { children, className, enableMasonry }: MasonryGalleryProps ) {
+	const ref = useRef< HTMLDivElement >( null );
+
+	useLayoutEffect( () => {
+		if ( ! ref.current || ! enableMasonry ) {
```

Comment:
Seems like we don't need the logic at all if `enableMasonry` is disabled. It could be possible to encapsulate the masonry logic into a separate component, and to not render that component at all if `enableMasonry` is false.

Reply:
If we were using the Masonry library, this might make sense. For the purposes of this component, though, I don't think it does. With or without `enableMasonry` we use the same `div` with the same class names and, to a large extent, the same CSS.